assets/javascripts/modules/addRemove.js

```diff
@@ -1,233 +1,166 @@
- *      <a href="#" class="add-remove__remove-btn" data-remove-btn>Delete</a>
- *    </script>
- *    <script type="text/template" data-add-btn-template>
- *      <a href="#" data-add-btn>Add another redirect URI</a>
- *    </script>
- *
- *    <ul data-add-remove-list>
- *      <li data-add-remove-item>
- *        <input type="text" value="Value 1" name="redirectUris[0]" data-add-remove-unique/>
- *      </li>
- *      <li data-add-remove-item>
- *        <input type="text" value="Value 2" name="redirectUris[1]" data-add-remove-unique/>
- *      </li>
- *      <li data-add-remove-item>
- *        <input type="text" value="Value 3" name="redirectUris[2]" data-add-remove-unique/>
- *      </li>
- *    </ul>
- *
- *  </div>
- *
+/*
+Add Remove
+
+A component to add and remove inputs on a page. You have the option of sending in custom attributes for:
+- `data-can-delete="true"` to add delete buttons
+- `data-delete-btn-text` to send in custom delete button text, this defaults to **Delete** and will only work when the
+ `data-can-delete="true"` is set
+- `data-add-btn-text` to send in custom add button text, this defaults to **Add another redirect URI**
+- `data-max` Maximum items allowed in the list
+
+Markup:
+<div data-add-remove
+     data-max="5"
+     data-can-delete="true",
+     data-delete-btn-text="Delete Me"
+     data-add-btn-text="Add Me">
+  <ul class="add-remove__list" data-add-remove-list>
+    <li class="add-remove__item" data-add-remove-item>
+      <input name="example[]"
+             type="text"
+             class="form-input input--medium"
+             data-add-remove-input />
+    </li>
+    <li class="add-remove__item" data-add-remove-item>
+      <input name="example[]"
+             type="text"
+             class="form-input input--medium"
+             data-add-remove-input />
+    </li>
+  </ul>
+</div>
  */
 
+var DEFAULT_ADD_BTN_TEXT = 'Add another redirect URI',
+    DEFAULT_DELETE_BTN_TEXT = 'Delete',
+    addRemoveContainer = '[data-add-remove]',
+    addRemoveList = '[data-add-remove-list]',
+    addRemoveItem = '[data-add-remove-item]',
+    addRemoveInput = '[data-add-remove-input]',
+    addButton = '[data-add-btn]',
+    maxItems = 'data-max';
 
-var addRemoveContainer = '[data-add-remove]',
-    addRemoveList      = '[data-add-remove-list]',
-    addRemoveItem      = '[data-add-remove-item]',
-    addButton          = '[data-add-btn]',
-    removeButton       = '[data-remove-btn]',
-    maxItems           = 'data-add-remove-max';
-
-var itemTemplate       = '[data-item-template]',
-    addBtnTemplate     = '[data-add-btn-template]',
-    removeBtnTemplate  = '[data-remove-btn-template]';
-
-
-/**
- * Init add/remove, where necessary
- *
- * @param
- */
-module.exports = function() {
-
+var init = function () {
   var $addRemoveContainers = $(addRemoveContainer);
 
-  // only continue if counter html exists
-  if($addRemoveContainers.length === 0) return;
+  if ($addRemoveContainers.length) {
```

Comment:
Style preference really, prefer conditionals with braces and the true conditional allows you in. Rather than the true conditional turning you away.
Yeah just styling really